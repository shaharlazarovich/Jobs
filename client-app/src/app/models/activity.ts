export interface IAttendee {
    userName: string;
    displayName: string;
    image: string;
    isHost: boolean;
    following?: boolean,
}

//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our activity array, in order to add a count
//of activities - which then, we could device in the number of pages
export interface IActivitiesEnvelope {
    activities: IActivity[];
    activityCount: number
}

export interface IActivity {
    id: string;
    title: string;
    description: string;
    category: string;
    date: Date;
    city: string;
    venue: string;
    isGoing: boolean;
    isHost: boolean;
    attendees: IAttendee[];
    comments: IComment[];
}

export interface IComment {
    id: string;
    createdAt: Date;
    body: string;
    username: string;
    displayName: string;
    image: string;
}

export interface IActivityFormValues extends Partial<IActivity> {
    time?: Date    
}

export class ActivityFormValues implements IActivityFormValues {
    id?: string = undefined;
    title: string = '';
    category: string = '';
    description: string = '';
    date?:Date = undefined;
    time?: Date = undefined;
    city: string = '';
    venue: string = '';

    constructor(init?: IActivityFormValues) {
        if (init && init.date) {
            init.time = init.date;
        }
        Object.assign(this, init); //this will initialize our object of class ActivityFormValues
                                  // with the above values
    }
}