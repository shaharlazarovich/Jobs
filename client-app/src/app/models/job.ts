//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our jobs array, in order to add a count
//of jobs - which then, we could device in the number of pages
export interface IJobsEnvelope {
    jobs: IJob[];
    jobsCount: number
}

export interface IJob {
    id: number;
    jobName: string;
    company: string;
    replication: string;
    servers: string;
    lastRun: Date;
    date: Date;
    time: Date;
    rta: string;
    results: string;
    key: string;
    rtoNeeded: string;
    jobIP: string;
}


export interface IJobFormValues extends Partial<IJob> {
     
}

export class JobFormValues implements IJobFormValues {
    id?: number = 0;
    jobName: string = '';
    company: string = '';
    replication: string = '';
    servers: string = '';
    lastRun?:Date = undefined;
    date?:Date = undefined;
    time?:Date = undefined;
    rta?: string = '';
    results: string = '';
    key: string = '';
    rtoNeeded: string = '';
    jobIP: string = '';

    constructor(init?: IJobFormValues) {
        Object.assign(this, init); //this will initialize our object of class JobFormValues
                                  // with the above values
    }
}