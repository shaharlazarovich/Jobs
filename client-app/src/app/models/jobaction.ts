//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our jobs array, in order to add a count
//of jobs - which then, we could device in the number of pages
export interface IJobActionsEnvelope {
    jobActions: IJobAction[];
    jobActionsCount: number
}

export interface IJobAction {
    id: string;
    jobId: string;
    jobName: string;
    userId: string;
    actionDate: Date;
    remoteIP: string;
    remoteResponse: string;
    requestProperties: string;
    source: string;
    action: string;
}

export class JobAction implements IJobAction {
    id: string = '';
    jobId: string = '';
    jobName: string = '';
    userId: string = '';
    actionDate: Date = new Date();
    remoteIP: string = '';
    remoteResponse: string = '';
    requestProperties: string = '';
    source: string = '';
    action: string = '';

    constructor(init?: IJobAction) {
        Object.assign(this, init); //this will initialize our object of class JobActions
                                  // with the above values
    }
}