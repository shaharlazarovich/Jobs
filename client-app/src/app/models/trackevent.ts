//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our jobs array, in order to add a count
//of jobs - which then, we could device in the number of pages
export interface ITrackEventsEnvelope {
    trackEvents: ITrackEvent[];
    trackEventsCount: number
}

export interface ITrackEvent {
    id: string;
    jobId?: string;
    jobName?: string;
    userId?: string;
    actionDate: Date;
    remoteIP?: string;
    remoteResponse?: string;
    requestProperties?: string;
    source: string;
    event: string;
}

export class TrackEvent implements ITrackEvent {
    id: string = '';
    jobId?: string = '';
    jobName?: string = '';
    userId?: string = '';
    actionDate: Date = new Date();
    remoteIP?: string = '';
    remoteResponse?: string = '';
    requestProperties?: string = '';
    source: string = '';
    event: string = '';

    constructor(init?: ITrackEvent) {
        Object.assign(this, init); //this will initialize our object of class JobActions
                                  // with the above values
    }
}