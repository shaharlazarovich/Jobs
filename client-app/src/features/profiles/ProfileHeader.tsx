import React from 'react';
import { Segment, Item, Header, Grid, Divider, Reveal } from 'semantic-ui-react';
import { IProfile } from '../../app/models/profile';
import { observer } from 'mobx-react-lite';

interface IProps {
    profile: IProfile,
    isCurrentUser: boolean,
    loading: boolean
}
const ProfileHeader: React.FC<IProps> = ({
    profile,
    isCurrentUser,
    loading
  }) => {
  return (
    <Segment data-test="component-profile-header">
      <Grid>
        <Grid.Column width={12}>
          <Item.Group>
            <Item>
              <Item.Image
                avatar
                size='small'
                src={profile.image || '/assets/user.png'}
              />
              <Item.Content verticalAlign='middle'>
                <Header as='h1'>{profile.displayName}</Header>
              </Item.Content>
            </Item>
          </Item.Group>
        </Grid.Column>
        <Grid.Column width={4}>
          <Divider/>
          {!isCurrentUser && 
          <Reveal animated='move'>
            <Reveal.Content visible style={{ width: '100%' }}>
              
            </Reveal.Content>
            <Reveal.Content hidden>
              
            </Reveal.Content>
          </Reveal>
          }
        </Grid.Column>
      </Grid>
    </Segment>
  );
};

export default observer(ProfileHeader);
