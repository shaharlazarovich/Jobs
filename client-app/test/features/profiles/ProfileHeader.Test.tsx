import { ReactWrapper } from "enzyme";
import React from "react";
import { IProfile } from '../../../src/app/models/profile'
import { findByTestAttr } from '../../common/testUtils';
import ProfileHeader from "../../../src/features/profiles/ProfileHeader";
import { profileMock } from "../../common/mocks";
import { setupWrapper } from "../../common/wrapper";

describe('Profile Header UniTest', () => {
    let wrapper: ReactWrapper;
    let testProfile: IProfile;
    beforeEach(async () => {
        testProfile = { ...profileMock };
        wrapper = await setupWrapper(<ProfileHeader profile={testProfile} isCurrentUser={true} />);
    });
    test('renders without error', () => {
        const component = findByTestAttr(wrapper, 'component-profile-header');
        expect(component.length).toBe(1);
    });
    test('should match the snapshot', () => {
        const component = findByTestAttr(wrapper, 'component-profile-header');
        expect(component).toMatchSnapshot();
    });
})