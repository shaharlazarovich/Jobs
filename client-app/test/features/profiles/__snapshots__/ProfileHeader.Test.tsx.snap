// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Header UniTest should match the snapshot 1`] = `
<Segment
  data-test="component-profile-header"
>
  <div
    className="ui segment"
    data-test="component-profile-header"
  >
    <Grid>
      <div
        className="ui grid"
      >
        <GridColumn
          width={12}
        >
          <div
            className="twelve wide column"
          >
            <ItemGroup>
              <div
                className="ui items"
              >
                <Item>
                  <div
                    className="item"
                  >
                    <ItemImage
                      avatar={true}
                      size="small"
                      src="/assets/user.png"
                    >
                      <Image
                        as="img"
                        avatar={true}
                        size="small"
                        src="/assets/user.png"
                        ui={true}
                        wrapped={true}
                      >
                        <div
                          className="ui small avatar image"
                        >
                          <img
                            src="/assets/user.png"
                          />
                        </div>
                      </Image>
                    </ItemImage>
                    <ItemContent
                      verticalAlign="middle"
                    >
                      <div
                        className="middle aligned content"
                      >
                        <Header
                          as="h1"
                        >
                          <h1
                            className="ui header"
                          >
                            fakeName
                          </h1>
                        </Header>
                      </div>
                    </ItemContent>
                  </div>
                </Item>
              </div>
            </ItemGroup>
          </div>
        </GridColumn>
        <GridColumn
          width={4}
        >
          <div
            className="four wide column"
          >
            <Divider>
              <div
                className="ui divider"
              />
            </Divider>
          </div>
        </GridColumn>
      </div>
    </Grid>
  </div>
</Segment>
`;
